flink-runtime/src/main/java/org/apache/flink/runtime/messages/checkpoint/DeclineCheckpoint.java

```diff
@@ -48,15 +43,24 @@ public DeclineCheckpoint(JobID job, ExecutionAttemptID taskExecutionId, long che
 	public DeclineCheckpoint(JobID job, ExecutionAttemptID taskExecutionId, long checkpointId, Throwable reason) {
 		super(job, taskExecutionId, checkpointId);
 
-		if (reason == null ||
-			reason.getClass() == AlignmentLimitExceededException.class ||
-			reason.getClass() == CheckpointDeclineOnCancellationBarrierException.class ||
-			reason.getClass() == CheckpointDeclineSubsumedException.class ||
-			reason.getClass() == CheckpointDeclineTaskNotCheckpointingException.class ||
-			reason.getClass() == CheckpointDeclineTaskNotReadyException.class ||
-			reason.getClass() == InputEndOfStreamException.class) {
-			// null or known common exceptions that cannot reference any dynamically loaded code
+		if (reason == null) {
```

Comment:
Why not instead just
```
		if (reason == null || reason instanceof CheckpointException) {
			this.reason = reason;
		} else {
			// some other exception. replace with a serialized throwable, to be on the safe side
			this.reason = new SerializedThrowable(reason);
		}
```
All `CheckpointException` objects should be from the Flink code and not user defined and therefore not require any `SerializedThrowable`